module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -821,7 +835,14 @@ public function getMyHolds($patron)
                 ) : '';
             $inTransit = $entry['status']['code'] === 't';
             $requestId = $this->extractId($entry['id']);
-            $updateDetails = ($available || $inTransit) ? '' : $requestId;
+            // Allow the user to attempt update if freezing is enabled or the hold
+            // is not available or in transit. Checking if the hold can be freezed
+            // up front is slow, so defer it to when update is requested.
+            if ($freezeEnabled || (!$available && !$inTransit)) {
+                $updateDetails = $requestId;
```

Comment:
I think you need to initialize $updateDetails before this, or have an else clause, to prevent it from being potentially undefined when used below.

Reply:
(Not sure why PHPstan didn't complain about this -- maybe I'm misreading something).

Reply:
Oops. And since it's used in a loop, the previous value might have been used. Now fixed.